spec/services/hwf_reference_generator_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe HwfReferenceGenerator, type: :service do
```

Comment:
I'm thinking that we should probably try to test the collision case as well. Perhaps if the actual reference number string was generated by a private method, let's say `reference_string`, then you can use RSpec stub to return different values each time:

```ruby
before do
  create(:online_submission, reference: 'collision')
  allow(ReferenceGenerator).to receive(:reference_string).and_return('collision', 'no-collision')
end
```

And afterwards test that it was the 'no-collision' winner. Something like that?